src/usbg_9pfs.rs

```diff
@@ -46,38 +55,30 @@ fn setup_9pfs_gadget(device: &String) -> Result<()> {
     mkdir("/sys/kernel/config/usb_gadget/9pfs/configs/c.1")?;
     mkdir("/sys/kernel/config/usb_gadget/9pfs/configs/c.1/strings/0x409")?;
 
-    let function = format!("/sys/kernel/config/usb_gadget/9pfs/functions/usb9pfs.{device}");
-    let link = format!("/sys/kernel/config/usb_gadget/9pfs/configs/c.1/usb9pfs.{device}");
+    let function = format!("/sys/kernel/config/usb_gadget/9pfs/functions/usb9pfs.{udc}");
+    let link = format!("/sys/kernel/config/usb_gadget/9pfs/configs/c.1/usb9pfs.{udc}");
     mkdir(&function)?;
     symlink(&function, &link)?;
 
-    debug!(
-        "Attaching 9pfs gatget to UDC {}",
-        udc.as_bytes().escape_ascii()
-    );
-    write_file(
-        "/sys/kernel/config/usb_gadget/9pfs/UDC",
-        udc.as_encoded_bytes(),
-    )?;
+    debug!("Attaching 9pfs gatget to UDC {udc}",);
+    write_file("/sys/kernel/config/usb_gadget/9pfs/UDC", &udc)?;
```

Comment:
Hmm, the file name read above is not  necessarily UTF-8, so converting it into a String is technically not correct.